Mozc for Android keeps crashing

Environment
-------------------------
Confirmed with Mozc 2.17.2287.102 running on Android 6.0.1 MNB29K / Nexus 5 2013.

Possible affected versions are from Mozc 2.17.2264.102 to Mozc 2.17.2287.102.
Probably there is no device/OS version dependency.

Steps to reproduce:
-------------------------
1. Build Mozc 2.17.2287.102 for Android.
2. Install Mozc into Android 6.0.1 MMB29K on Nexus 5 2013.
3. Enable Mozc.
4. Switch to Mozc.

Expected behavior
-------------------------
Mozc starts working.

Actual behavior
-------------------------
Mozc starts crashing.

Here is the relevant crash log.
```
01-01 07:01:38.536  3870  3870 I Mozc    : We use local Mozc engine.
01-01 07:01:38.555  3870  3901 E Mozc    : loadLibrary failed
01-01 07:01:38.555  3870  3901 E Mozc    : java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZN4mozc3jni19JavaHttpClientProxy9SetJavaVMEP7_JavaVM" referenced by "/data/app/org.mozc.android.inputmethod.japanese-1/lib/arm/libmozc.so"...
01-01 07:01:38.555  3870  3901 E Mozc    : 	at java.lang.Runtime.loadLibrary(Runtime.java:372)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at java.lang.System.loadLibrary(System.java:1076)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at org.mozc.android.inputmethod.japanese.session.MozcJNI.load(MozcJNI.java:71)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at org.mozc.android.inputmethod.japanese.session.LocalSessionHandler.initialize(LocalSessionHandler.java:97)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at org.mozc.android.inputmethod.japanese.session.SessionExecutor$ExecutorMainCallback.handleMessage(SessionExecutor.java:321)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at android.os.Handler.dispatchMessage(Handler.java:98)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at android.os.Looper.loop(Looper.java:148)
01-01 07:01:38.555  3870  3901 E Mozc    : 	at android.os.HandlerThread.run(HandlerThread.java:61)
--------- beginning of crash
01-01 07:01:38.555  3870  3901 E AndroidRuntime: FATAL EXCEPTION: Session worker thread
01-01 07:01:38.555  3870  3901 E AndroidRuntime: Process: org.mozc.android.inputmethod.japanese, PID: 3870
01-01 07:01:38.555  3870  3901 E AndroidRuntime: java.lang.RuntimeException: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZN4mozc3jni19JavaHttpClientProxy9SetJavaVMEP7_JavaVM" referenced by "/data/app/org.mozc.android.inputmethod.japanese-1/lib/arm/libmozc.so"...
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at org.mozc.android.inputmethod.japanese.session.MozcJNI.load(MozcJNI.java:75)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at org.mozc.android.inputmethod.japanese.session.LocalSessionHandler.initialize(LocalSessionHandler.java:97)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at org.mozc.android.inputmethod.japanese.session.SessionExecutor$ExecutorMainCallback.handleMessage(SessionExecutor.java:321)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:148)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:61)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZN4mozc3jni19JavaHttpClientProxy9SetJavaVMEP7_JavaVM" referenced by "/data/app/org.mozc.android.inputmethod.japanese-1/lib/arm/libmozc.so"...
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at java.lang.Runtime.loadLibrary(Runtime.java:372)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at java.lang.System.loadLibrary(System.java:1076)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	at org.mozc.android.inputmethod.japanese.session.MozcJNI.load(MozcJNI.java:71)
01-01 07:01:38.555  3870  3901 E AndroidRuntime: 	... 5 more
```

Additional Information:
-------------------------
This is a regression accidentally introduced by google/mozc@9cd4d36d2d51968bfad369fc6ee7703594007aef. The root cause is that ```base/base.gyp:jni_proxy``` GYP target is not linked to ```libmozc.so``` unless ```branding=="GoogleJapaneseInput"```

--- a/src/android/android.gyp
+++ b/src/android/android.gyp
@@ -576,6 +576,7 @@
       'product_dir': '<(abs_android_dir)/libs/<(abi)',
       'dependencies': [
         '../base/base.gyp:base',
+        '../base/base.gyp:jni_proxy',
         '../dictionary/dictionary.gyp:dictionary',
         '../engine/engine.gyp:engine_factory',
         '../session/session.gyp:session',

--- a/src/mozc_version_template.txt
+++ b/src/mozc_version_template.txt
@@ -1,6 +1,6 @@
 MAJOR=2
 MINOR=17
-BUILD=2287
+BUILD=2288
 REVISION=102
 # NACL_DICTIONARY_VERSION is the target version of the system dictionary to be
 # downloaded by NaCl Mozc.

--- a/src/net/net.gyp
+++ b/src/net/net.gyp
@@ -83,11 +83,6 @@
                 'http_client_pepper.cc',
               ],
             }],
-            ['target_platform=="Android"', {
-              'dependencies': [
-                '../base/base.gyp:jni_proxy'
-              ],
-            }],
           ],
         }, {  # blanding!=GoogleJapaneseInput
           'sources': [

